server/service/src/sync/api/error.rs

```diff
@@ -1,58 +1,128 @@
 use super::*;
 use reqwest::{Response, StatusCode, Url};
-use serde::Deserialize;
+use serde::{
+    de::{value::StrDeserializer, IntoDeserializer},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
 use thiserror::Error;
+use url::ParseError;
 
 #[derive(Error, Debug)]
-pub enum SyncApiError {
-    #[error("status: ({status}) error: {source}")]
-    MappedError {
-        source: SyncErrorV5,
-        status: StatusCode,
-    },
-    #[error("Error connecting to server: ({url})")]
-    ConnectionError { source: reqwest::Error, url: Url },
-    #[error("{0}")]
-    ResponseParsingError(ParsingResponseError),
-    #[error("{0}")]
-    Other(#[from] anyhow::Error),
+#[error("Sync api error, url: '{url}', route: '{route}'")]
```

Comment:
I'm not seeing this formatting when testing. Perhaps doing it wrong?

The error provides plenty of information - it can be pretty dense to read, with the nested errors though. Misspelling a domain gives:

```
2022-10-03 13:42:30.215860 INFO service::sync::site_info - Requesting site info
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: RequestSiteInfoError(SyncApiError { source: ConnectionError(reqwest::Error { kind: Request, url: Url { scheme: "http", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("localhostie")), port: Some(2048), path: "/sync/v5/site", query: None, fragment: None }, source: hyper::Error(Connect, ConnectError("dns error", Custom { kind: Uncategorized, error: "failed to lookup address information: nodename nor servname provided, or not known" })) }), url: Url { scheme: "http", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("localhostie")), port: Some(2048), path: "/", query: None, fragment: None }, route: "/sync/v5/site" })', /Users/m2/GitHub/open-msupply/server/server/src/lib.rs:129:22
```

I was wondering if the Url could be formatted with `Into<string>` instead of the whole object?

Reply:
This is definitely not what I intended as output, please see comment below: https://github.com/openmsupply/open-msupply/pull/671#issuecomment-1264868340. 🙏 for catching this